### PR TITLE
Transition CI from Azure Pipelines to GitHub Actions

### PR DESCRIPTION
**Pull request by altheaden**
In this PR, I've switched the CI workflows from azure pipelines to GH actions. There's a build workflow for testing code, and a docs workflow for deploying the documentation via the `gh-pages` branch. This change allows the CI for MPAS-Analysis to be more in line with our CI procedures in other repos (e.g., Polaris, Mache, Compass).

### CODE REVIEW

**Comment by altheaden** [comment]
@xylar If you would like, I can add a pr template to this PR as well. Or, I could add that when I make my dependabot branch (or not at all), up to you.

File: .github/workflows/docs_workflow.yml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+          # gh-pages branch must already exist
+          git clone https://github.com/MPAS-Dev/MPAS-Analysis.git --branch gh-pages --single-branch gh-pages
+          # Make sure we're in the gh-pages directory.
+          cd gh-pages
+          # Create `.nojekyll` (if it doesn't already exist) for proper GH Pages configuration.
+          touch .nojekyll
+          # Add `index.html` to point to the `develop` branch automatically.
+          printf '<meta http-equiv="refresh" content="0; url=./develop/index.html" />' > index.html
+          # Only replace `develop` docs with latest changes. Docs for releases should be untouched.
+          rm -rf develop
```
Comment (xylar) [inline .github/workflows/docs_workflow.yml:79]: @altheaden, as we discussed, let's make sure `develop` here becomes the name of the target branch.  See if you can figure that out based on the link I sent you today. If not, no worries.  We can look it up together.  The redirect should still be for `develop` regardless of the target branch, though.

File: .github/workflows/docs_workflow.yml [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+          cd docs
+          # gh-pages branch must already exist
+          git clone https://github.com/MPAS-Dev/MPAS-Analysis.git --branch gh-pages --single-branch gh-pages
+          # Make sure we're in the gh-pages directory.
+          cd gh-pages
+          # Create `.nojekyll` (if it doesn't already exist) for proper GH Pages configuration.
+          touch .nojekyll
+          # Add `index.html` to point to the `develop` branch automatically.
+          printf '<meta http-equiv="refresh" content="0; url=./develop/index.html" />' > index.html
+          # Only replace `${{ github.head_ref || github.ref_name }}` docs with latest changes. Docs for releases should be untouched.
```
Comment (xylar) [inline .github/workflows/docs_workflow.yml:78]:
```suggestion
          # Only replace docs in a directory with the destination branch name with latest changes. Docs for releases should be untouched.
```

**Comment by xylar** [comment]
Thanks so much @altheaden!  I may need to clean up the docs after this goes in.  Let's see...

**Pull request merged**